.changeset/changelog-stylelint.js

```diff
@@ -5,9 +5,28 @@
 const { getInfo, getInfoFromPullRequest } = require('@changesets/get-github-info');
 
 /**
- * @type {import('@changesets/types').ChangelogFunctions}
+ * @typedef { 'major' | 'minor' | 'patch' } ReleaseType
+ * @typedef { Record<ReleaseType, Array<Promise<string>>> } ReleaseLines
+ * @typedef { Record<ReleaseType, string[]> } ResolvedReleaseLines
+ * @type {import('@changesets/types').ChangelogFunctions & { reorderReleaseLines(releaseLines: ReleaseLines): Promise<ResolvedReleaseLines> }}
  */
 const changelogFunctions = {
+	async reorderReleaseLines(releaseLines) {
+		const resolved = {
+			major: await Promise.all(releaseLines.major),
+			minor: await Promise.all(releaseLines.minor),
+			patch: await Promise.all(releaseLines.patch),
+		};
+
+		return Object.entries(resolved).reduce(
+			(acc, [type, lines]) =>
+				Object.assign(acc, {
+					// TODO: implement https://stylelint.io/maintainer-guide/pull-requests/#merging
```

Comment:
I'm still not quite sure how should I implement this order algorithm, @jeddy3 @ybiquitous can you offer some help here?

Reply:
If it's hard to implement the order, it seems that we can manually fix the generated changelog entries. What do you think?

Reply:
It's not hard, I just don't know what should I write.  😅

Reply:
@ybiquitous I don't understand

> Order the item within the group by the widest-reaching first to the smallest, and then alphabetically by rule name.

What does this mean? Sorry, I'm not a native English speaker.

Reply:
Can you help to review whether my latest changes are expected?

Reply:
Could you let us know how to test this change locally?

Reply:
@ybiquitous 

1. generate some changesets files with `npx changeset`
2. run `npm run version` and check CHANGELOG file and `package.json`

Reply:
I've tested it locally just to run the `npm run version` command, and then the diff is below:

```diff
diff --git a/.changeset/nice-lemons-reflect.md b/.changeset/nice-lemons-reflect.md
deleted file mode 100644
index 6f75e4c2c..000000000
--- a/.changeset/nice-lemons-reflect.md
+++ /dev/null
@@ -1,5 +0,0 @@
----
-"stylelint": minor
----
-
-Added: `declaration-block-no-duplicate-properties` autofix
diff --git a/CHANGELOG.md b/CHANGELOG.md
index d957c5985..e3cd7e630 100644
--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,5 +1,11 @@
 # Changelog
 
+## 14.12.0
+
+### Minor Changes
+
+- Added: `declaration-block-no-duplicate-properties` autofix ([#6296](https://github.com/stylelint/stylelint/pull/6296)) ([@fpetrakov](https://github.com/fpetrakov)).
+
 ## 14.11.0
 
 - Added: `ignoreAfterCombinators: []` to `selector-max-universal` ([#6275](https://github.com/stylelint/stylelint/pull/6275)).
```

The `### Minor Changes` section seems extra, but we can manually remove it on a release pull request.

Reply:
As the [maintainer guide describes](https://stylelint.io/maintainer-guide/pull-requests/#merging) as below,

> Prefix the item with either: "Removed", "Changed", "Deprecated", "Added", or "Fixed".

Is it possible to sort change entries by order of `"Removed", "Changed", "Deprecated", "Added", or "Fixed"` (importance order)?

- `"Removed"`, `"Changed"` - major (breaking)
- `"Deprecated"`, `"Added"` - minor
- `"Fixed"` - patch

Reply:
I remembered you said, but I didn't find the source: we can endure these changes. And the reorder algorithm only works in sections.

Besides, such section titles are much easier for reading IMO.

Even if we want to rollback, we can just remove these section title.

Reply:
Personally, I don't think changesets will officially support this, but fork to support our purpose is very simple just like our patch right now.

Reply:
I also don't quite sure why we're focusing these unimportant things.🤣

Reply:
I have an idea how to implement it just now, I'll try ASAP.